### PR TITLE
metric: add more lint rules for metric panels

### LINKED ISSUE #58435: define more lint rules for tidb metric panels
## Enhancement
existing rules include:
- panel ID must be unique, see https://github.com/pingcap/tidb/pull/51568
- there is no `.*` around `$tidb_cluster`, see https://github.com/pingcap/tidb/pull/54135

rules we need to add:
- the `datasource` field should be `${DS_TEST-CLUSTER}`, i guess TiUP uses this to replace with real datasource
- panel must have a `title`
- `row` type panel should be collapsed initially

rules we add to follow a consistent style:
- height of most panels is 7, but some uses 6 or 8/9, we unify them to 7
- first char of words in titles are capitalized in most panels, such as `Duration 95 By Instance`, it doesn't follow english grammar, but it's already used too much, we follow it anyway

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #58435

Problem Summary:

### What changed and how does it work?
see issue
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [x] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/metrics/grafana/overview.json [hunk 1]
```diff
@@ -253,6 +253,8 @@
               "refId": "M"
             }
           ],
+          "title": "Probe Success",
```
Comment (D3Hunter) [inline pkg/metrics/grafana/overview.json:256]: this and below panel don't have a `title`, not sure what's it used for, I add a temporary title based on the expression

File: pkg/metrics/grafana/performance_overview.json
Comment (D3Hunter) [inline pkg/metrics/grafana/performance_overview.json:1]: it's a large change because the first row is open previously

**Review by hawkingrei — APPROVED** [review]

File: pkg/metrics/grafana/tidb_resource_control.json
Comment (jackysp) [inline pkg/metrics/grafana/tidb_resource_control.json:1]: You should not modify this file directly, you should edit [tidb/pkg/metrics/grafana/tidb_summary.jsonnet at master · pingcap/tidb](https://github.com/pingcap/tidb/blob/master/pkg/metrics/grafana/tidb_summary.jsonnet) and [tidb/pkg/metrics/grafana/tidb_resource_control.jsonnet at master · pingcap/tidb](https://github.com/pingcap/tidb/blob/master/pkg/metrics/grafana/tidb_resource_control.jsonnet)

File: pkg/metrics/grafana/tidb_summary.json
Comment (jackysp) [inline pkg/metrics/grafana/tidb_summary.json:1]: ditto, and see https://github.com/pingcap/tidb/blob/master/pkg/metrics/grafana/README.md for more details.

**Review by jackysp — APPROVED** [review]
LGTM

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**